// @project-sunbird/telemetry-sdk/gruntfile.js
const fs = require('fs');
module.exports = function(grunt) {
    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        version: '3.2.',
        karma: {
            telemetryV3: {
                configFile: 'karma.renderer.config.js',
            },
            telemetryFunV3: {
                configFile: 'karma.telemetry.fun.config.js',
            }

        },
        concat: {
            telemetryLib: {
                src: [
                    './libs/ajv.min.js',
                    './schema/telemetry-spec.js',
                    './libs/detectClient.js',
                    './libs/md5.js',
                    './libs/ua-parser.min.js',
                    './libs/fingerprint2.min.js',
                    './core/telemetryV3Interface.js',
                    './core/telemetrySyncManager.js',
                ],
                dest: './dist/index.js'
            }
        }
    });
    grunt.loadNpmTasks('grunt-karma');
    grunt.loadNpmTasks('grunt-contrib-uglify');
    grunt.loadNpmTasks('grunt-contrib-copy');
    grunt.loadNpmTasks('grunt-contrib-clean');
    grunt.loadNpmTasks('grunt-contrib-concat');
    grunt.registerTask('build-telemetry-lib', ['concat:telemetryLib']);
    grunt.registerTask('renderer-telemetryV3', ['karma:telemetryV3']);
    grunt.registerTask('telemetry-lib-test', ['karma:telemetryFunV3']);

};
